test/scaladoc/run/inlineToStr-strips-unwanted-text.scala

```diff
@@ -0,0 +1,59 @@
+import scala.tools.nsc.doc.model._
+import scala.tools.partest.ScaladocModelTest
+
+object Test extends ScaladocModelTest {
+
+  override def code = """
+  /** This comment contains ^superscript^ */
+  class Foo {
+    /** This comment contains ,,subscript,, */
+    def bar = ???
+
+    /** This comment contains a link [[https://scala.epfl.ch/]] */
+    def baz = ???
+
+    /** This comment contains an <strong>html tag</strong> */
+    def qux = ???
+
+    /** This comment contains a<br> single html tag */
+    def quux = ???
+  }
+  """
+  def scaladocSettings = ""
+
+  def testModel(root: Package) = {
+    import scala.tools.nsc.doc.base.comment._
+    import access._
+
+    def inlineToStr(inl: Inline): String = inl match {
+      case Chain(items) => items flatMap (inlineToStr(_)) mkString ""
+      case Italic(in) => inlineToStr(in)
+      case Bold(in) => inlineToStr(in)
+      case Underline(in) => inlineToStr(in)
+      case Superscript(in) => inlineToStr(in)
+      case Subscript(in) => inlineToStr(in)
+      case Link(raw, title) => inlineToStr(title)
+      case Monospace(in) => inlineToStr(in)
+      case Text(text) => text
+      case Summary(in) => inlineToStr(in)
+      case HtmlTag(tag) => "<[^>]*>".r.replaceAllIn(tag, "")
+      case EntityLink(Text(text), _) => text
+    }
```

Comment:
Copy-pasting is not ideal, obviously.. Given that this function doesn't depend on the class `Page` in which it is defined, how about pulling it out into the companion object? I don't see a problem with that function being public (vs. protected).